### ISSUE TITLE
no-extra-parens ignores multiline jsx return 

### ISSUE THREAD

**Issue by dropfen** [eslint/eslint#7952]
```
{
   "no-extra-parens": ["error", "all"]
}

```

**What did you do? Please include the actual source code causing the issue.**

```js

class MyComp extends React.Component {
  render() {
    return (
      <div>
        <p>One</p>
        <p>Two</p>
      </div>
    );
  }
}

```

**What did you expect to happen?**

Should see an error / warning because of the parantheses after return.

**What actually happened? Please include the actual, raw output from ESLint.**

nothnig

**Comment by mysticatea** [comment]
Thank you for the report.

However, this is intentional behavior.
`no-extra-parens` allows parentheses at the location of `[no LineTerminator here]` in [the language spec](https://www.ecma-international.org/ecma-262/7.0/#prod-ReturnStatement) because deletion of the parentheses changes semantics.

```js
class MyComp extends React.Component {
  render() {
    return // return undefined!
      <div>
        <p>One</p>
        <p>Two</p>
      </div>;
  }
}
```

**Comment by dropfen** [comment]
@mysticatea Sure, the team (i'am working in currently) wants it this way:

```
return <div>
  <div />
</div>
```

**Comment by mysticatea** [comment]
Hm, I feel that it's not the responsibility of `no-extra-parens` rule, need `multiline-jsx-style` rule or something like.
How about `"never"` option of [react/jsx-wrap-multilines](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-wrap-multilines.md) rule? (though, the option does not exist currently...)

**Comment by dropfen** [comment]
thank you, I've opened an issue there https://github.com/yannickcr/eslint-plugin-react/issues/1035
